### PR TITLE
[MRG] Nistats integration

### PR DESCRIPTION
**Pull request by mrahim**
Addresses #172 
A first attempt to replace nipy by nistats.
The nipy dependency and sub dependencies are replaced by pandas only.
The glm functions keep the same structure, the nipy `DesignMatrix` is replaced by `DataFrame`.
The impact is on the reporter, the examples and the continuous integration.

CircleCI is running successfully 3 examples that are not too long.
https://circleci.com/gh/mrahim/pypreprocess/190

Travis is okay too.
https://travis-ci.org/mrahim/pypreprocess/builds/92285255

### CODE REVIEW

**Comment by dohmatob** [comment]
Gr8. I'll do a review asap.

**Comment by dohmatob** [comment]
Subject stats report generation broken:

before
before

after
after

**Comment by mrahim** [comment]
Exactly, since `paradigm` is a `DataFrame`, we should tweak a little bit the report generator.

**Comment by bthirion** [comment]
Looks great. Note that Nistats is going to evolve. We'll probably have to resync regularly.

Only negative comment: The Haxby example takes ages on my laptop -- which is arguably not the right equipment to run this.

**Comment by mrahim** [comment]
@bthirion : Yes, there is a bug in that script, it is taking the bold image instead of the anat.

**Comment by dohmatob** [comment]
Ideally, that haxby fix should have gone in a separate PR since it has nothing to do with nistats integration per se.

File: examples/easy_start/nipype_preproc_spm_auditory.py [hunk 1]
```diff
@@ -9,12 +9,14 @@
 import numpy as np
 import matplotlib.pyplot as plt
 import nibabel
-from nipy.modalities.fmri.experimental_paradigm import BlockParadigm
-from nipy.modalities.fmri.design_matrix import make_dmtx
-from nipy.modalities.fmri.glm import FMRILinearModel
 from pypreprocess.nipype_preproc_spm_utils import do_subjects_preproc
 from pypreprocess.datasets import fetch_spm_auditory
 from pypreprocess.reporting.glm_reporter import generate_subject_stats_report
+from pandas import DataFrame
```
Comment (dohmatob) [inline examples/easy_start/nipype_preproc_spm_auditory.py]: It more common to do (like with `numpy`): `import pandas as pd` and then use `pd.Dataframe`.

File: examples/easy_start/nipype_preproc_spm_auditory.py [hunk 2]
```diff
@@ -48,12 +53,12 @@
...
-                          paradigm, hrf_model=hrf_model,
-                          drift_model=drift_model, hfcut=hfcut)
+design_matrix = make_design_matrix(
+    frametimes, paradigm, hrf_model=hrf_model, drift_model=drift_model,
+    period_cut=hfcut)
 
 # plot and save design matrix
-ax = design_matrix.show()
+ax = plot_design_matrix(design_matrix)
 ax.set_position([.05, .25, .9, .65])
```
Comment (dohmatob) [inline examples/easy_start/nipype_preproc_spm_auditory.py]: At this point, the user is over the pen with the code complexity.This line looks like magic. Users shouldn't have to do such manual tweaking.

File: examples/easy_start/nipype_preproc_spm_auditory.py (hunk 1 above)
Comment (bthirion) [inline examples/easy_start/nipype_preproc_spm_auditory.py]: Agreed, this si going to be fixed in Nilearn. I think we should not worry about that  in pypreprocess.

File: examples/easy_start/nipype_preproc_spm_auditory.py (hunk 2 above)
Comment (bthirion) [inline examples/easy_start/nipype_preproc_spm_auditory.py]: Agreed. I think we should consider fixing this in nistats.

**Comment by dohmatob** [comment]
LGTM (the PR is better than the current code quality in master...). Once comments are addressed, we can merge (and have the awesome ci implemented by @mehdi as bonus).

**Pull request merged**